### ISSUE TITLE
CKAN 2.5.3 with ckan.root_path returns internal server errors

### ISSUE THREAD

**Issue by MrkGrgsn** [ckan/ckan#3306]
### CKAN Version if known (or site URL)
2.5.3

### Please describe the expected behaviour
Pages load.

### Please describe the actual behaviour
When using a non-root `ckan.root_path` all pages fail to load with an internal server error.

```
[Sun Nov 06 23:24:27.992034 2016] [wsgi:error] [pid 24567] [remote 127.0.0.1:124]   File "/usr/lib/ckan/25/src/ckan/ckan/lib/helpers.py", line 198, in url_for
[Sun Nov 06 23:24:27.992349 2016] [wsgi:error] [pid 24567] [remote 127.0.0.1:124]     return _add_i18n_to_url(my_url, locale=locale, **kw)
[Sun Nov 06 23:24:27.992358 2016] [wsgi:error] [pid 24567] [remote 127.0.0.1:124]   File "/usr/lib/ckan/25/src/ckan/ckan/lib/helpers.py", line 294, in _add_i18n_to_url
[Sun Nov 06 23:24:27.992370 2016] [wsgi:error] [pid 24567] [remote 127.0.0.1:124]     root_path = re.sub('{{LANG}}', locale, root_path)
[Sun Nov 06 23:24:27.992377 2016] [wsgi:error] [pid 24567] [remote 127.0.0.1:124]   File "/usr/lib64/python2.7/re.py", line 155, in sub
[Sun Nov 06 23:24:27.992462 2016] [wsgi:error] [pid 24567] [remote 127.0.0.1:124]     return _compile(pattern, flags).sub(repl, string, count)
[Sun Nov 06 23:24:27.992471 2016] [wsgi:error] [pid 24567] [remote 127.0.0.1:124]   File "/usr/lib64/python2.7/re.py", line 286, in _subx
[Sun Nov 06 23:24:27.992483 2016] [wsgi:error] [pid 24567] [remote 127.0.0.1:124]     template = _compile_repl(template, pattern)
[Sun Nov 06 23:24:27.992489 2016] [wsgi:error] [pid 24567] [remote 127.0.0.1:124]   File "/usr/lib64/python2.7/re.py", line 271, in _compile_repl
[Sun Nov 06 23:24:27.992498 2016] [wsgi:error] [pid 24567] [remote 127.0.0.1:124]     p = sre_parse.parse_template(repl, pattern)
[Sun Nov 06 23:24:27.992505 2016] [wsgi:error] [pid 24567] [remote 127.0.0.1:124]   File "/usr/lib64/python2.7/sre_parse.py", line 737, in parse_template
[Sun Nov 06 23:24:27.992640 2016] [wsgi:error] [pid 24567] [remote 127.0.0.1:124]     s = Tokenizer(source)
[Sun Nov 06 23:24:27.992649 2016] [wsgi:error] [pid 24567] [remote 127.0.0.1:124]   File "/usr/lib64/python2.7/sre_parse.py", line 192, in __init__
[Sun Nov 06 23:24:27.992660 2016] [wsgi:error] [pid 24567] [remote 127.0.0.1:124]     self.__next()
[Sun Nov 06 23:24:27.992666 2016] [wsgi:error] [pid 24567] [remote 127.0.0.1:124]   File "/usr/lib64/python2.7/sre_parse.py", line 194, in __next
[Sun Nov 06 23:24:27.992678 2016] [wsgi:error] [pid 24567] [remote 127.0.0.1:124]     if self.index >= len(self.string):
[Sun Nov 06 23:24:27.992693 2016] [wsgi:error] [pid 24567] [remote 127.0.0.1:124] TypeError: object of type 'Locale' has no len()
```

### What steps can be taken to reproduce the issue? 
Plain source install of 2.5.3 using apache 2.4 w/mod_wsgi and `ckan.root_path = /data`

**Comment by MrkGrgsn** [comment]
I think this was a missed backport as it was fixed previously in master 22bf30ff6c45a43f1e71ee247b427874e5ab9f52

**Comment by amercader** [comment]
You are correct, we need to get this in the next patch release
